app/services/external_api/va_dot_gov_service.rb

```
@@ -0,0 +1,156 @@
+require "json"
+
+class ExternalApi::VADotGovService
+  class << self
+    # :nocov:
+    def get_distance(point, ids)
+      page = 1
+      facility_results = []
+
+      until ids.empty?
+        results = fetch_facilities_with_ids(
+          query: { lat: point[0], long: point[1], page: page },
+          ids: ids
+        )
+
+        ids -= results[:facilities].pluck(:id)
+        facility_results += results[:facilities]
+
+        break if !results[:has_next]
+        page += 1
+        sleep 1
+      end
+
+      Rails.logger.info("Unable to find api.va.gov facility data for: #{ids.join(', ')}.") if !ids.empty?
```

Do we want to throw an error or something here? If this is a case that should never happen, we probably want to know about it.

---

the code shouldn't fail if there are a handful of ids that aren't found, but we do want to know about it so we can check the data. would we get an alert from `Rails.logger.error`?

---

We could throw a sentry error without failing. Here's code that sends a sentry error: https://github.com/department-of-veterans-affairs/caseflow/blob/bf90bb2acd8219dc46b36c73eb09a63cd79e6f6f/app/controllers/application_controller.rb#L12

And we wouldn't have to fail then. Though I would counter that maybe failing is good here. If we don't fail, we might not show the closest facility to the HS coordinator.

---

ok. failing works for me then.